starknet_py/net/full_node_client.py

```diff
@@ -703,7 +719,11 @@ async def simulate_transactions(
         # pylint: disable=too-many-arguments
         """
         Simulates a given sequence of transactions on the requested state, and generates the execution traces.
-        If one of the transactions is reverted, raises CONTRACT_ERROR.
+        Note that some of the transactions may revert, in which case no error is thrown, but revert details can be seen
+        on the returned trace object.
+        Note that some of the transactions may revert, this will be reflected by the revert_error property in the trace.
+        Other types of failures (e.g. unexpected error or failure in the validation phase) will result
+        in TRANSACTION_EXECUTION_ERROR.
```

Comment:
could be consolidated probably?

Reply:
Basically this is copied from [the spec](https://github.com/starkware-libs/starknet-specs/blob/v0.6.0/api/starknet_trace_api_openrpc.json#L42), but I can try to make it more concise.

Reply:
Let's, this reads quite badly imo